twenty-first/src/util_types/mmr/shared_advanced.rs

```diff
@@ -144,11 +142,35 @@ pub fn get_authentication_path_node_indices(
     }
 }
 
-/// Return a list of the peak heights for a given leaf count
+/// Return a list of the peak heights for a given leaf count.
+///
+/// # Examples
+///
+/// ```
+/// # use twenty_first::util_types::mmr::shared_advanced::get_peak_heights;
+/// assert_eq!(get_peak_heights(0b1010), vec![3, 1]);
+/// assert_eq!(get_peak_heights(0b1011), vec![3, 1, 0]);
+/// ```
 pub fn get_peak_heights(leaf_count: u64) -> Vec<u8> {
-    // The peak heights in an MMR can be read directly from the bit-decomposition
-    // of the leaf count.
-    indices_of_set_bits(leaf_count)
+    // In an MMR, the peak heights directly correspond the leaf count's bit decomposition. That is,
```

Comment:
Eht this rewrite? Otherwise this PR looks good.

Reply:
Main reason because the used method was only ever used once, across all our crates. This made the corresponding MMR method very shallow. Citing [46531c4](https://github.com/Neptune-Crypto/twenty-first/pull/192/commits/46531c4a7b6071856ae0937382887c09a04e57eb):

> Function `indices_of_set_bits` seemed like it could be useful as a general function. However, in practice, it is only used in Merkle Mountain Ranges.

That said, I'm not married to this change. :relieved: